### PR TITLE
fix #170 must be the same user

### LINKED ISSUE #170: Add validation to question that 'question.user' and 'question.survey.user' have to be the same

### PR DESCRIPTION
**Pull request by danielnottingham**
fix #170 

# Add Appropriate Title
**Models:**
- Add method survey_must_have_same_user

**Model:**
- 'is valid when question and survey have the same user'
- 'is not valid when question and survey have different user'

#### Only select the appropriate.
- [x] **Model testing.**
- [ ] **Request testing.**
- [ ] **System testing.**
- [ ] **No tests required.**

### CODE REVIEW

File: backend/spec/models/question_spec.rb [hunk 1]
```diff
@@ -76,4 +76,20 @@
...
+
+  describe 'validate survey same user' do
+    it 'is valid when question and survey have the same user' do
+      survey_teacher = create(:survey, user: user_teacher)
+      question = create(:question, user: user_teacher, survey: survey_teacher)
+
+      expect(question).to be_valid
+    end
+
+    it 'is not valid when question and survey have different user' do
```
Comment (edimossilva) [inline backend/spec/models/question_spec.rb:88]: add spec to check the error message in the model

**Review by edimossilva — CHANGES_REQUESTED** [review]
I will fix the factory soon

File: backend/spec/models/question_spec.rb (hunk 1 above)
Comment (danielnottingham) [reply backend/spec/models/question_spec.rb:88]: Ok

File: backend/app/models/question.rb [hunk 2]
```diff
@@ -21,6 +22,12 @@ class Question < ApplicationRecord
 
   private
 
+  def survey_must_have_same_user
+    return unless user&.present? && survey&.present?
+
+    errors.add(:question, 'Must be the same user!') if user != survey.user
```
Comment (edimossilva) [inline backend/app/models/question.rb:28]: andd translations

**Review by edimossilva — CHANGES_REQUESTED** [review]

**Review by edimossilva — APPROVED** [review]
Niceee, squash and merge!

**Pull request merged**